bluesky/tests/test_examples.py

```diff
@@ -342,6 +345,106 @@ def sim_kill():
     assert stop - start > 2
 
 
+def test_caught_pause_in_flyer(RE):
+    from ophyd.status import Status
+
+    class MyFlyer:
+        name = "flyer"
+
+        def prepare(self, start_from):
+            status = Status()
+            self.start_from = start_from
+            status.set_finished()
+            return status
+
+        def kickoff(self):
+            status = Status()
+            status.set_finished()
+            return status
+
+        def complete(self):
+            self.complete_status = Status()
+            return self.complete_status
+
+        def pause(self):
+            # Let the plan handle it
+            raise RefusePause()
+
+        def resume(self):
+            pass
+
+    flyers = [MyFlyer(), MyFlyer()]
+    msgs = []
+    RE.msg_hook = msgs.append
+
+    def do_scan(start_from: int):
+        for flyer in flyers:
+            yield from bps.prepare(flyer, start_from, group="prepare", wait=False)
+        yield from bps.wait(group="prepare")
+        for flyer in flyers:
+            yield from bps.kickoff(flyer, group="kickoff", wait=False)
+        yield from bps.wait(group="kickoff")
+        for flyer in flyers:
+            yield from bps.complete(flyer, group="complete", wait=False)
+        yield from bps.wait(group="complete")
+        # Just to check we got to the end
+        yield from bps.null()
+
+    @bpp.run_decorator()
+    def pausing_plan():
+        done = False
+        start_from = 1
+        while not done:
+            try:
+                yield from do_scan(start_from)
+            except FailedPause:
+                yield from bps.checkpoint()
```

Comment:
@tacaswell I missed something here, I've swallowed the pause at the plan level here to allow me to checkpoint and break out, but then I immediately continue on. I guess instead I should wait until the RE says resume. How do I do that?

Reply:
add a `yield from bps.pause()` after the check point.

We need to make sure that `obj.pause()` is called on _everything_ before we raise